velox/docs/develop/spilling.rst

```diff
@@ -504,4 +545,51 @@ spilling:
   partition number goes first.
 * To parallelize the hash table build from the spilled partition, the hash join
   bridge will split the spill partition files among the hash build operators
-  with each one having an equally-sized shard to restore.g
+  with each one having an equally-sized shard to restore.
+
+Future Work
+-----------
+Memory Arbitration
+^^^^^^^^^^^^^^^^^^
+Augment the memory management with memory arbitration support which kicks in to
+reclaim memory from running queries when any operator fails to allocate or
+reserve memory. The memory arbitrator can reclaim memory from both spillable
+and non-spillbale operators. For spillable operator, we need to add arena
+compaction to free up unused memory chunks. For non-spillable operator such as
+partial aggregation, the memory arbitrator can reclaim memory by triggering the
+partial aggregation state flush. The memory arbitration not only prevents a
```

Comment:
The memory arbitration logic will allow queries to complete successfully using limited amounts of memory and enable dynamic memory sharing between concurrent queries to improve overall memory efficiency.